src/librustc/traits/trans/mod.rs

```diff
@@ -0,0 +1,73 @@
+use dep_graph::{DepGraph, DepNode, DepTrackingMap, DepTrackingMapConfig};
+use hir::def_id::DefId;
+use std::cell::RefCell;
+use std::marker::PhantomData;
+use traits::Vtable;
+use ty::{self, Ty};
+
+/// Specializes caches used in trans -- in particular, they assume all
+/// types are fully monomorphized and that free regions can be erased.
```

Comment:
Yeah, probably it's true that these caches aren't so important, though I'm not sure I'm inclined to change anything here in this particular PR. (I did however intend to move a lot of the routines that manipulate said caches out of trans and into this file, and I forgot to do so.)